### PR TITLE
[SDK Reference Book #1]use spring team's yaml data as datasource

### PR DESCRIPTION
**Pull request by wangmingliang-ms**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-sdk-reference-book/src/main/java/com/microsoft/azure/toolkit/intellij/azuresdk/model/AzureSdkArtifactEntity.java [hunk 1]
```diff
@@ -5,41 +5,52 @@
...
-    @JsonProperty("PlannedVersions")
-    private String plannedVersions;
-    @JsonProperty(value = "Hide")
-    private Boolean isHide;
-    @JsonProperty("Notes")
-    private String notes;
+    private List<Link> links;
+
+    public String generateMavenDependencySnippet(String version) {
+        return String.join("", "",
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-sdk-reference-book/src/main/java/com/microsoft/azure/toolkit/intellij/azuresdk/model/AzureSdkArtifactEntity.java:30]: one more empty string?

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-sdk-reference-book/src/main/java/com/microsoft/azure/toolkit/intellij/azuresdk/service/AzureSdkLibraryService.java [hunk 2]
```diff
@@ -5,98 +5,47 @@
...
 
 public class AzureSdkLibraryService {
-    private static final ObjectMapper mapper = new CsvMapper();
-    private static final String SDK_METADATA_URL = "https://raw.githubusercontent.com/Azure/azure-sdk/master/_data/releases/latest/java-packages.csv";
-
-    @Getter
-    private List<AzureSdkServiceEntity> serviceEntities = new ArrayList<>();
-    @Getter
-    private List<AzureSdkArtifactEntity> artifactEntities = new ArrayList<>();
+    private static final ObjectMapper mapper = new YAMLMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-sdk-reference-book/src/main/java/com/microsoft/azure/toolkit/intellij/azuresdk/service/AzureSdkLibraryService.java:22]: If service update the schema, shall we skip the unknow properties exception? Or library service will break until we fix the model.

Comment (wangmingliang-ms) [reply PluginsAndFeatures/azure-toolkit-for-intellij/azure-sdk-reference-book/src/main/java/com/microsoft/azure/toolkit/intellij/azuresdk/service/AzureSdkLibraryService.java:22]: just skip.
this config is to skip unused properties.

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-sdk-reference-book/src/main/java/com/microsoft/azure/toolkit/intellij/azuresdk/model/AzureSdkArtifactEntity.java (hunk 1 above)
Comment (wangmingliang-ms) [reply PluginsAndFeatures/azure-toolkit-for-intellij/azure-sdk-reference-book/src/main/java/com/microsoft/azure/toolkit/intellij/azuresdk/model/AzureSdkArtifactEntity.java:30]: removed.

**Review by Flanker32 — APPROVED** [review]

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-sdk-reference-book/src/main/java/com/microsoft/azure/toolkit/intellij/azuresdk/model/AzureSdkArtifactEntity.java [hunk 3]
```diff
@@ -5,41 +5,42 @@
...
+        return String.join("",
+                "<dependency>\n",
+                "    <groupId>", this.groupId, "</groupId>\n",
+                "    <artifactId>", this.artifactId, "</artifactId>\n",
+                "    <version>", version, "</version>\n",
+                "</dependency>"
+        );
+    }
+
+    public String getLink(String rel) {
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-sdk-reference-book/src/main/java/com/microsoft/azure/toolkit/intellij/azuresdk/model/AzureSdkArtifactEntity.java:37]: rel -> linkType?

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-sdk-reference-book/src/main/java/com/microsoft/azure/toolkit/intellij/azuresdk/model/AzureSdkArtifactEntity.java [hunk 4]
```diff
@@ -5,41 +5,42 @@
...
-@ToString
 @Getter
+@Setter
 @NoArgsConstructor
+@JsonIgnoreProperties(ignoreUnknown = true)
 public class AzureSdkArtifactEntity {
-    @JsonProperty("Package")
-    private String packageName;
-    @JsonProperty("GroupId")
+    private String artifactId;
```
Comment (Flanker32) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-sdk-reference-book/src/main/java/com/microsoft/azure/toolkit/intellij/azuresdk/model/AzureSdkArtifactEntity.java:20]: It seem that the raw data also provide compatibility data for libraries, is this a backlog feature?

Comment (wangmingliang-ms) [reply PluginsAndFeatures/azure-toolkit-for-intellij/azure-sdk-reference-book/src/main/java/com/microsoft/azure/toolkit/intellij/azuresdk/model/AzureSdkArtifactEntity.java:20]: the full feature actually has 2 sub features:
* sdk reference book(readonly)
* dependency management

dependency compatibility is part of dependency management, which is not planned in this sprint.

**Pull request merged**